source/extensions/omni.isaac.lab/omni/isaac/lab/utils/buffers/circular_buffer.py

```diff
@@ -109,7 +121,7 @@ def append(self, data: torch.Tensor):
         # at the fist call, initialize the buffer
         if self._buffer is None:
             self._pointer = -1
-            self._buffer = torch.empty((self.max_length, *data.shape), dtype=data.dtype, device=self._device)
+            self._buffer = torch.zeros((self.max_length, *data.shape), dtype=data.dtype, device=self._device)
```

Comment:
@jtigue-bdai  Isn't it better to initialize the buffer to the latest data on the first append or right after a reset?

Otherwise, we assume that zero is a valid value for the observation, which may not always be true.
For example, if we are gathering a history of, lets say,  the gravitational force, which may be defined as  a strictly negative value, having a buffer filled with zeros as previous observations may not be within the expected distribution.

I would suggest that, during reset or at first init, all the indices in the history is initialized to the most recent data, possibly the one being passed into `append()`

Reply:
Thats a good point. I think you are right about filling on the first append.